x-pack/legacy/plugins/siem/public/components/paginated_table/index.tsx

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  EuiBasicTable,
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiPanel,
+  EuiPopover,
+} from '@elastic/eui';
+import { isEmpty, noop, getOr } from 'lodash/fp';
+import React, { memo, useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import { Direction } from '../../graphql/types';
+import { AuthTableColumns } from '../page/hosts/authentications_table';
+import { HeaderPanel } from '../header_panel';
+import { LoadingPanel } from '../loading';
+
+import * as i18n from './translations';
+
+const DEFAULT_DATA_TEST_SUBJ = 'paginated-table';
+
+export interface ItemsPerRow {
+  text: string;
+  numberOfRow: number;
+}
+
+export interface SortingBasicTable {
+  field: string;
+  direction: Direction;
+  allowNeutralSort?: boolean;
+}
+
+export interface Criteria {
+  page?: { index: number; size: number };
+  sort?: SortingBasicTable;
+}
+
+declare type HostsTableColumns = [
```

Comment:
nit -> maybe this can move to the hosts table

Reply:
Right now its only in there for the tests. Once i implement for hosts table I will move it.